lib/chef/http.rb

```diff
@@ -131,8 +131,8 @@ def post(path, json, headers={})
     #
     # === Parameters
     # path:: path part of the request URL
-    def delete(path, headers={})
-      request(:DELETE, path, headers)
+    def delete(path, json, headers={})
```

Comment:
this changes the API and breaks the world with dozens of

```
ArgumentError:
        wrong number of arguments (1 for 2..3)
```

errors in travis.  either this needs to be an optional argument or a new 'delete_with_body' method.